virttest/utils_misc.py

```diff
@@ -2765,20 +2747,56 @@ def get_all_disks_did(session, partition=False):
     :return: a dict with all disks did lists each include disk
              kname, serial and wwn.
     """
-    disks = list_linux_guest_disks(session, partition)
-    logging.debug("Disks detail: %s" % disks)
-    all_disks_did = {}
-    for line in disks:
-        kname = line.split('/')[2]
-        get_disk_info_cmd = "udevadm info -q property -p /sys/block/%s" % kname
-        output = session.cmd_output_safe(get_disk_info_cmd)
-        re_str = r"(?<=DEVNAME=/dev/)(.*)|(?<=ID_SERIAL=)(.*)|"
-        re_str += "(?<=ID_SERIAL_SHORT=)(.*)|(?<=ID_WWN=)(.*)"
-        did_list_group = re.finditer(re_str, output, re.M)
-        did_list = [match.group() for match in did_list_group if match]
-        all_disks_did[kname] = did_list
+    list_disk_cmd = "lsblk -o KNAME,SIZE,TYPE,MOUNTPOINT"
+    output = session.cmd_output_safe(list_disk_cmd)
+    devs = output.splitlines()
+    disks_list = {}
+    mount_list = []
+    for dev in devs:
+        dev = dev.split()
+        if "disk" in dev:
+            disks_list[dev[0]] = dev
+        if partition and "part" in dev:
+            disks_list[dev[0]] = dev
+        if len(dev) == 4:
+            mount_list.append(dev[0])
+            mount_disk = "".join(re.findall(r"([vhs]d[a-z])", dev[0]))
+            if mount_disk not in mount_list:
+                mount_list.append(mount_disk)
```

Comment:
Seems the if logic has some issue, how about following ones, the lines 2766 ~ 2769 can be deleted.
        if "disk" in dev:
            disks_list[dev[0]] = dev
        elif partition and "part" in dev:
            parted_disk = "".join(re.findall(r"([vhs]d[a-z])", dev[0]))
            if parted_disk not in disks_list.keys():
                disks_list[parted_disk] = dev
        if mounted and len(dev) == 4:
            mounted_disk = "".join(re.findall(r"([vhs]d[a-z])", dev[0]))
            if mounted_disk not in disks_list.keys():
                disks_list[mounted_disk] = dev